core/src/main/scala/org/apache/spark/status/api/v1/ApiRootResource.scala

```diff
@@ -222,6 +222,7 @@ private[spark] object ApiRootResource {
 private[spark] trait UIRoot {
   def getSparkUI(appKey: String): Option[SparkUI]
   def getApplicationInfoList: Iterator[ApplicationInfo]
+  def getApplicationInfoList(limit: Int): Iterator[ApplicationInfo]
```

Comment:
1. The null check occurs at the endpoint though so it doesn't affect this function. 
2. take(-1) and take(0) don't return the same thing, any negative input to take returns the whole list, this was checked in my previous pr and it is the default value in `historypage.js`